cluster/sbin/snappy-lead.sh

```diff
@@ -36,12 +37,43 @@ shift
 . "$SNAPPY_HOME/bin/load-spark-env.sh"
 . "$SNAPPY_HOME/bin/load-snappy-env.sh"
 
-
+noOfInputsArgs=$#
 
 # Start up  the lead instance
 function start_instance {
   "$SNAPPY_HOME"/bin/snappy leader "$mode" "$@"
 }
 
+if [ $noOfInputsArgs -le 1 ]
```

Comment:
Why are we assuming that noOfArgs le 1 will be the only case where -dir option might not be provided.
May the user gives other options and the list becomes greater than one and still the -dir option is missing.

Reply:
> Why are we assuming that noOfArgs le 1 will be the only case where -dir option might not be provided.
Resolved. Checking with all provided arguments
> May the user gives other options and the list becomes greater than one and still the -dir option is missing.
Resolved: Modified the code, now as per the current modification, if the user doesn't provide -dir argument, ask the user to provide it.